### PR TITLE
PAF-39: add navigation design

### PR DESCRIPTION
**Pull request by Rhodine-orleans-lindsay**
## What
- Create get-section behaviour to configure navigation menu links
- Remove next property from index.js for pages preceeding section pages as they were preventing this pages from being reached (the next property for these pages have now been configured in the get-section behaviour)
- Add views for layout, side-nav and main content
- Add css for side navigation
- Amend css for links so disabled links do not have govuk-link class
- Add get-section behaviour to hof.settings so it is applied across all pages 
- Add behaviours in hof.settings to server.js
- Add relevant tests

## Why
- Part of PAF form

## Testing
- tests passing locally

### CODE REVIEW

File: apps/paf/behaviours/get-section.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
+module.exports = superclass => class extends superclass {
```
Comment (sulthan-ahmed) [inline apps/paf/behaviours/get-section.js:1]: Is there a reason you used the word set rather than get? Is this file setting the steps and backlinks based on the new navigation design? I wonder if it's worth renaming the behaviour to something like `set-navigation-section`

File: apps/paf/behaviours/get-section.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    const locals = super.locals(req, res);
+    const crimeType = req.sessionModel.get('crime-type');
+    const immmigrationCrime = req.sessionModel.get('immigration-crime-group');
+    const smugglingCrime = req.sessionModel.get('smuggling-crime-group');
+    const crimeChildren = req.sessionModel.get('crime-children');
+
+    // enable section links once required fields have been completed
+    if (crimeType && (immmigrationCrime || smugglingCrime) && crimeChildren) {
+      locals.enableSection = true;
+      return locals;
```
Comment (sulthan-ahmed) [inline apps/paf/behaviours/get-section.js:122]: this is great 👍

File: assets/scss/app.scss [hunk 3]
```diff
@@ -10,7 +10,7 @@ ul {
   @extend .list-bullet;
 }
 
-a {
+a:link {
```
Comment (sulthan-ahmed) [inline assets/scss/app.scss:13]: what does this do?

File: hof.settings.json [hunk 4]
```diff
@@ -1,6 +1,9 @@
 {
   "appName":"Public Allegations Form",
   "theme": "govUK",
+  "behaviours": [
+    "./apps/paf/behaviours/get-section"
+  ],
```
Comment (sulthan-ahmed) [inline hof.settings.json:6]: remind me, this adds this behaviour to all the steps?

File: assets/scss/app.scss (hunk 3 above)
Comment (Rhodine-orleans-lindsay) [reply assets/scss/app.scss:13]: The `:link` property applies only to links with an href. Initially the govuk styling for links was assigned to anything with an anchor tag but because some of the menu links are disabled until required fields are completed,  I changed it so that the styling is only applied to enabled links i.e.those with an href.

File: apps/paf/behaviours/get-section.js (hunk 1 above)
Comment (Rhodine-orleans-lindsay) [reply apps/paf/behaviours/get-section.js:1]: Yeah, you're right;  it is setting properties based on the navigation rather than getting the actual section so I will amend the name

File: test/_unit/behaviours/get-section.spec.js [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+      res = reqres.res();
+      req.sessionModel.attributes.steps = [];
+
+      const GetSectionController = Behaviour(Controller);
+      controller = new GetSectionController({ template: 'index', route: '/index' });
+      controller._configure(req, res, done);
+    });
+
+
+    describe(`sets the 'next' property for pages preceeding the first page of each section`, () => {
```
Comment (sulthan-ahmed) [inline test/_unit/behaviours/get-section.spec.js]: This is good just maybe a bit wordy.  I was taught to write test names in simple narrative form and less technical so that anyone can understand especially junior developers. Maybe rewrite the describe as
describe(`sets the next page according to the section`)

File: test/_unit/behaviours/get-section.spec.js [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+        };
+
+        sandbox.stub(Controller.prototype, 'configure').yieldsAsync();
+      });
+
+      afterEach(() => {
+        sandbox.restore();
+      });
+
+      it(`the crime-location page has the 'next' property as '/report-person'`, () => {
```
Comment (sulthan-ahmed) [inline test/_unit/behaviours/get-section.spec.js]: this is good maybe write it as `next page` and below

**Review by sulthan-ahmed — COMMENTED** [review]
Fantastic work as always, just some minor suggestions no big deal at all 🙂

File: assets/scss/app.scss (hunk 3 above)
Comment (sulthan-ahmed) [reply assets/scss/app.scss:13]: Ace 👍

File: apps/paf/behaviours/get-section.js (hunk 1 above)
Comment (sulthan-ahmed) [reply apps/paf/behaviours/get-section.js]: Great, now it makes more sense to me 🙂

File: hof.settings.json (hunk 4 above)
Comment (sulthan-ahmed) [reply hof.settings.json:6]: nice, I think there's another way too. It might be in firearms

File: test/_unit/behaviours/get-section.spec.js (hunk 6 above)
Comment (Rhodine-orleans-lindsay) [reply test/_unit/behaviours/get-section.spec.js]: thanks Sulthan, I have amended those as suggested

File: test/_unit/behaviours/get-section.spec.js [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
+        req.form.options.route = '/are-you-eighteen';
+        controller.configure(req, res, () => {
+          const areYouEighteenNext = req.form.options.steps['/are-you-eighteen'].next;
+          areYouEighteenNext.should.deep.equal('/confirm');
+          req.form.options.next.should.deep.equal('/confirm')
+        })
+      });
+    });
+
+    describe('sets the backlink property for first page of each section', () => {
```
[the remark this answers is not in the document]
Comment (Rhodine-orleans-lindsay) [reply test/_unit/behaviours/get-section.spec.js]: I have amended those as suggested now, thanks

File: test/_unit/behaviours/get-section.spec.js (hunk 5 above)
Comment (Rhodine-orleans-lindsay) [reply test/_unit/behaviours/get-section.spec.js]: I have amended this as suggested now, thanks
